src/hydrate/platform/proxy-host-element.ts

```diff
@@ -97,27 +97,32 @@ export function proxyHostElement(elm: d.HostElement, cstr: d.ComponentConstructo
           enumerable: true,
         });
 
-        // instance
-        Object.defineProperty((cstr as any).prototype, memberName, {
-          get: function (this: any) {
-            if (origGetter && attrPropVal === undefined && !getValue(this, memberName)) {
-              // if the initial value comes from an instance getter
-              // the element will never have the value set. So let's do that now.
-              setValue(this, memberName, origGetter.apply(this), cmpMeta);
-            }
-
-            // if we have a parsed value from an attribute / or userland prop use that first.
-            // otherwise if we have a getter already applied, use that.
-            const ref = getHostRef(this);
-            return ref.$instanceValues$?.get(memberName) !== undefined
-              ? ref.$instanceValues$?.get(memberName)
-              : origGetter
-                ? origGetter.apply(this)
-                : getValue(this, memberName);
-          },
-          configurable: true,
-          enumerable: true,
-        });
+        if (!(cstr as any).prototype.__stencilAugmented) {
```

Comment:
Should we move the section:
```ts
    // instance prototype should only be processed once
    (cstr as any).prototype.__stencilAugmented = true;
```
into the if block?

Reply:
I don't think so because the if block is within a loop running over all the class' props.
Setting the flag within the loop will mean only the first class member will get augmented.

Reply:
Thanks for pointing that out.